lib/elixir/pages/anti-patterns/design-anti-patterns.md

```diff
@@ -5,7 +5,46 @@ play within a codebase.
 
 ## Primitive obsession
 
-TODO
+#### Problem
+
+This anti-pattern can be felt when Elixir basic types (for example, *integer*, *float*, and *string*) are abusively used in function parameters and code variables, rather than creating specific composite data types (for example, *structs*, and *map*) that can better represent a domain.
+
+#### Example
+
+An example of this anti-pattern is the use of a single *string* to represent an `Address`. An `Address` is a more complex structure than a simple basic (aka, primitive) value.
+
+```elixir
+defmodule Foo do    
+  @spec bar(String.t()) :: any()
+  def bar(address) do
+    # Do something with address...
+  end  
+end
+```
```

Comment:
````suggestion
```

Another example of this anti-pattern is using floating numbers to model money and currency, when [richer data structures should be preferred](https://hexdocs.pm/ex_money/).

````